solutions/4-getting-started-js/event_queue.js

```diff
@@ -0,0 +1,34 @@
+var queue = (function () {
+  const data = {}
+  const publicMethods = {}
+
+  publicMethods.on = function (eventName, callback) {
+    if (eventName in data) {
+      data[eventName].push(callback)
+    } else {
+      data[eventName] = [callback]
+    }
+  }
+
+  publicMethods.remove = function (eventName) {
+    data.remove(eventName)
+  }
+
+  publicMethods.trigger = function (eventName) {
+    while (data[eventName].length !== 0) {
+      (data[eventName].shift())()
```

Comment:
I believe you're modifying the `data`. The next time you call .trigger(), there'll be no callbacks?